_posts/2021/2021-08-02-dont-call-replacement-classes-new.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Don't call a new version of something 'new' when refactoring"
+description: "A opinionated naming subtlety"
+layout: article
+category: ruby
+image:
+  base: '2021/dont-call-replacement-classes-new'
+  alt: "New"
+  credit: Nick Fewings
+  source: "https://unsplash.com/photos/1SsUquHPNT8"
+
+---
+
+In long-lived applications, we often want to rewrite or refactor an approach that we have previously taken.
+
+Here's a, non-obvious, naming technique I've used during refactoring to ensure that the improved code ends up in well named classes and any old implementation is clearly marked for deletion.
+
+You might find this helps organise your work or communicate the intent of future changes with your teammates.
```

Comment:
*that this helps to organise...or better communicate...

Reply:
```suggestion
You might find this helps to organise your work or better communicate the intent of future changes with your teammates.
```